debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/YugabyteReplicaIdentityIT.java

```diff
@@ -0,0 +1,350 @@
+                             .with(PostgresConnectorConfig.SNAPSHOT_MODE, PostgresConnectorConfig.SnapshotMode.NEVER.getValue())
+                             .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                             .build();
+    start(PostgresConnector.class, config);
+    assertConnectorIsRunning();
+
+    // YB Note: Added a wait for replication slot to be active.
+    TestHelper.waitFor(Duration.ofSeconds(10));
+
+    waitForAvailableRecords(10_000, TimeUnit.MILLISECONDS);
+    // there shouldn't be any snapshot records
+    assertNoRecordsToConsume();
+
+    // insert and verify 2 new records
+    TestHelper.execute("INSERT INTO s2.a VALUES (1, 22, 'random text value');");
+    TestHelper.execute("UPDATE s2.a SET aa = 12345 WHERE pk = 1;");
+
+    SourceRecords actualRecords = consumeRecordsByTopic(2);
+    List<SourceRecord> records = actualRecords.recordsForTopic(topicName("s2.a"));
+
+    SourceRecord insertRecord = records.get(0);
+    SourceRecord updateRecord = records.get(1);
+
+    YBVerifyRecord.isValidInsert(insertRecord, PK_FIELD, 1);
+    YBVerifyRecord.isValidUpdate(updateRecord, PK_FIELD, 1);
+
+    Struct updateRecordValue = (Struct) updateRecord.value();
+    assertThat(updateRecordValue.get(Envelope.FieldName.AFTER)).isNotNull();
+    assertThat(updateRecordValue.get(Envelope.FieldName.BEFORE)).isNull();
+
+    // After field will have entries for all the columns.
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("pk").getInt32("value")).isEqualTo(1);
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("aa").getInt32("value")).isEqualTo(12345);
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("bb").getString("value")).isEqualTo("random text value");
+  }
+
+  @Test
+  public void shouldProduceEventsWithValuesForChangedColumnWithReplicaIdentityChange() throws Exception {
+    // YB Note: Note that even if we do not alter, the default replica identity on service is CHANGE.
+    TestHelper.execute("ALTER TABLE s2.a REPLICA IDENTITY CHANGE;");
+
+    Configuration config = TestHelper.defaultConfig()
+                             .with(PostgresConnectorConfig.SNAPSHOT_MODE, PostgresConnectorConfig.SnapshotMode.NEVER.getValue())
+                             .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                             .build();
+    start(PostgresConnector.class, config);
+    assertConnectorIsRunning();
+
+    // YB Note: Added a wait for replication slot to be active.
+    TestHelper.waitFor(Duration.ofSeconds(10));
+
+    waitForAvailableRecords(10_000, TimeUnit.MILLISECONDS);
+    // there shouldn't be any snapshot records
+    assertNoRecordsToConsume();
+
+    // insert and verify 2 new records
+    TestHelper.execute("INSERT INTO s2.a VALUES (1, 22, 'random text value');");
+    TestHelper.execute("UPDATE s2.a SET aa = 12345 WHERE pk = 1;");
```

Comment:
Can you also add an Update where we set one of the columns to null. This will verify our logic for unchanged toasted datum for nulls.